compose/config/types.py

```diff
@@ -118,58 +119,96 @@ def normalize_paths_for_engine(external_path, internal_path):
     """Windows paths, c:\my\path\shiny, need to be changed to be compatible with
     the Engine. Volume paths are expected to be linux style /c/my/path/shiny/
     """
-    if not IS_WINDOWS_PLATFORM:
-        return external_path, internal_path
 
-    if external_path:
-        drive, tail = os.path.splitdrive(external_path)
+    def _normalize(path):
+        drive, tail = splitdrive(path)
 
         if drive:
-            external_path = '/' + drive.lower().rstrip(':') + tail
+            path = '/' + drive.lower().rstrip(':') + tail
+
+        return path.replace('\\', '/')
+
+    if external_path:
+        external_path = _normalize(external_path)
+
+    return external_path, _normalize(internal_path)
```

Comment:
Is `normalize_paths_for_engine()` still a useful function? Seems like we could just move `_normalize()` up to the top level (maybe renaming it to `normalize_path_for_engine()`, singular) and invoke it directly from `VolumeSpec._parse_win32()`.